.gitignore

```diff
@@ -12,7 +12,7 @@ Makefile.in
 aclocal.m4
 autom4te.cache
 output/
-/build/
+/build/autoconf/
```

Comment:
Ignoring the entire build seemed correct to me: we have build/default and such

Reply:
I added a build/make/java.mk file. Hence, modified this to ignore the only other folder inside build/ similar to the openweave organization.
Would you suggest I move this java.mk file to a different spot?

Reply:
Have moved the java.mk file outside of build folder, thus, voiding the need for this change.
This is done in https://github.com/project-chip/connectedhomeip/pull/1454 from my fork off of the public CHIP repo (this PR is from before, when the repo was private and updating there is turning out to be more of a hassle).